SSL Support (#2500)
confluentinc/ksql: docs/installation/server-config/security.rst

.. include:: ../../../../includes/installation-types-zip-tar.rst

.. _config-ksql-for-https:

> Configuring KSQL for HTTPS

Author: Meh, it's fine as it is IMHO.

--------------------------
KSQL can be configured to use HTTPS rather than the default HTTP for all communication.

If you haven't already, you will need to :ref:`create SSL key and trust stores <generating-keys-certs>`.

Use the following settings to configure the KSQL server to use HTTPS:

::

    listeners=https://hostname:port
    ssl.keystore.location=/var/private/ssl/ksql.server.keystore.jks
    ssl.keystore.password=xxxx
    ssl.key.password=yyyy

Note the use of the HTTPS protocol in the ``listeners`` config.

To enable the server to authenticate clients (2-way authentication), use the following additional
settings:

::

    ssl.client.auth=required
    ssl.truststore.location=/var/private/ssl/ksql.server.truststore.jks
    ssl.truststore.password=zzzz

.. _configuring-cli-for-https:

-----------------------------
Configuring the CLI for HTTPS
-----------------------------
If the KSQL server is configured to use HTTPS, CLI instances may need to be configured with
suitable key and trust stores.

> If the SSL certificate the server is using is not signed by a recognised public Certificate Authority

Reviewer: ```suggestion
If the SSL certificate the server is using is not signed by a recognised public Certificate Authority,
```

Author: done.

the CLI will need to be configured with a trust store that trusts the servers SSL certificate.

If you haven't already, you will need to :ref:`create SSL key and trust stores <generating-keys-certs>`.

Use the following settings to configure the CLI server:

::

    ssl.truststore.location=/var/private/ssl/ksql.client.truststore.jks
    ssl.truststore.password=zzzz

If the server is performing client authentication (2-way authentication), use the following
additional settings:

::

    ssl.keystore.location=/var/private/ssl/ksql.client.keystore.jks
    ssl.keystore.password=xxxx
    ssl.key.password=yyyy

Settings for the CLI can be stored in a suitable file and passed to the CLI via the ``--config-file``
command-line arguments, for example:

.. code:: bash

>     <ksql-install>bin/ksql --config-file ./config/ksql-cli.properties http://localhost:8088

Author: Good catch!

Configuring KSQL for Basic HTTP Authentication
----------------------------------------------
KSQL can be configured to require users to authenticate using a username and password via the Basic
HTTP authentication mechanism.

> .. note:: If using Basic authentication it is highly recommended that you

Reviewer: ```suggestion
.. note:: If you're using Basic authentication, we recommended that you
```

Author: done.
